lib/virt_autotest/utils.pm

```diff
@@ -105,23 +105,56 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
-#check host to make sure it works well
+#check kernels by grep keywords from journals
 #welcome everybody to extend this function
-sub check_host {
+sub check_failures_in_journal {
+    my $machine //= 'localhost';
+
+    my $all_failures = "";
+    my @warnings = ('Started Process Core Dump', 'Call Trace');
+    foreach my $warn (@warnings) {
+        my $cmd = "journalctl | grep '$warn'";
+        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
+        if (script_run($cmd) == 0) {
+            if (get_var('KNOWN_KERNEL_BUG')) {
+                my $failure = "$warn on $machine: " . get_var('KNOWN_KERNEL_BUG');
+                $all_failures .= $failure;
+                record_soft_failure("$failure (Please ignore the 'fake bsc#5555' that has to be written here due to the silly PR code rule)");
+            }
+            else {
+                record_soft_failure("$warn on $machine: Fake bsc#5555. This is an unknown failure which need to be investigated! (Please ignore the 'fake bsc#5555' that has to be written here due to the silly PR code rule)");
+                $all_failures .= "$warn on $machine: Unknown failure";
+            }
+        }
+    }
+    return $all_failures;
+}
 
+# Do some basic check to host to see if it is working well
+# Return 'pass' and 'fail' if there are or are not failures.
+# Welcome everybody to extend this function
+sub check_host_health {
+    my $failures = check_failures_in_journal;
+    if ($failures eq "") {
+        record_info("Host good!");
+        return 'pass';
+    }
+    record_info("Failures in journals are found", $failures, result => 'fail');
+    return 'fail';
 }
 
-#check guest to make sure it works well
-#welcome everybody to extend this function
-sub check_guest {
+# Do some basic check to specified guest tto see if it is working well
+# Return 'pass' and 'fail' if there are or are not failures.
+# Welcome everybody to extend this function
+sub check_guest_health {
     my $vm = shift;
 
     #check if guest is still alive
     validate_script_output "virsh domstate $vm", sub { /running/ };
-
-    #TODO: other checks like checking journals from guest
-    #need check the oops bug
-
+    my $failures = check_failures_in_journal($vm);
+    return 'fail' unless $failures eq "";
```

Comment:
```
return 'fail' if $failures;
```

Reply:
It seems you did not record unhealthy guest in record_info ?

Reply:
No, I did not add the red warning for guest failures. I think the soft_fail is ok. check_guest is supposed to be called by test owners on their own. They can decide if they need a red mark in their test. In the final validate_system_health test module, the result will be 'fail' if there are guest failures.